src/main/java/de/hysky/skyblocker/skyblock/garden/VisitorHelper.java

```diff
@@ -87,10 +92,18 @@ private static void processVisitorItem(String visitorName, ScreenHandler handler
         if (visitorItem == null || !visitorItem.contains(DataComponentTypes.LORE) || ItemUtils.getLoreLineIf(visitorItem, t -> t.contains("Times Visited")) == null) return;
         ItemStack acceptButton = handler.getSlot(29).getStack();
         if (acceptButton == null) return;
-        processLore(visitorName, ItemUtils.getLore(acceptButton));
+        processLore(visitorName, getTextureOrNull(visitorItem), ItemUtils.getLore(acceptButton));
     }
 
-    private static void processLore(String visitorName, List<Text> loreList) {
+    private static @Nullable String getTextureOrNull(ItemStack stack) {
+        if (!stack.isOf(Items.PLAYER_HEAD) || stack.get(DataComponentTypes.PROFILE) == null) return null;
+        return stack.get(DataComponentTypes.PROFILE).properties().get("textures").stream()
+                .map(Property::value)
+                .findFirst()
+                .orElse(null);
+    }
```

Comment:
This could be a helper in `ItemUtils` along with the other usages of this code, but we will need to fix that once this has been merged into 1.20.5.